tools/run-tests.py

```diff
@@ -195,6 +195,49 @@ def run_jsc_stress(engine, arch):
         env={'PYTHONPATH': '.'})
 
 
+def _run_regression_tests(engine, files, is_fail, assert_js):
+    fails = 0
+    for file in files:
+        passed = True
+
+        proc = Popen([engine, assert_js, file], stdout=PIPE)
+        out, _ = proc.communicate()
+
+        if is_fail and proc.returncode or not is_fail and not proc.returncode:
+            print('%sOK: %s%s' % (COLOR_GREEN, file, COLOR_RESET))
+        else:
+            print('%sFAIL(%d): %s%s' % (COLOR_RED, proc.returncode, file, COLOR_RESET))
+            print(out)
+
+            fails += 1
+
+    return fails
+
+
+@runner('regression-tests', default=True)
+def run_regression_tests(engine, arch):
+    REGRESSION_DIR = join(PROJECT_SOURCE_DIR, 'test', 'regression-tests')
+    REGRESSION_XFAIL_DIR = join(REGRESSION_DIR, 'xfail')
+    REGRESSION_ASSERT_JS = join(REGRESSION_DIR, 'assert.js')
+
+    print('Running regression tests:')
+    xpass = glob(join(REGRESSION_DIR, 'issue-*.js'))
+    xpass_result = _run_regression_tests(engine, xpass, False, REGRESSION_ASSERT_JS)
+
+    print('Running regression tests expected to fail:')
+    xfail = glob(join(REGRESSION_XFAIL_DIR, '*.js'))
```

Comment:
Although assert.js is not in this dir, it is more consistent to use issue-*.js for xfails, too.